proposal/first-class-mixins.md

```diff
@@ -0,0 +1,292 @@
+
+* If the value is not being inspected, throw an error.
+
+* Otherwise:
+
+  * Emit `"get-mixin("`.
+
+  * Emit a double quote (`"`), then the `name` of the mixin, then another double
+    quote.
+
+  * Emit `")"`.
+
+## Functions
+
+### `meta.type-of()`
+
+Add the following clause to the [`meta.type-of()`] function and the top-level
+`type-of()` function:
+
+[`meta.type-of()`]: ../spec/built-in-modules/meta.md#type-of
+
+* If `$value` is a mixin, return an unquoted string with value `"mixin"`.
+
+### `meta.get-mixin()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.get-mixin($name, $module: null)
+```
+
+* If `$name` is not a string, throw an error.
+
+* If `$module` is null:
+
+  * Return the result of resolving a mixin named `$name`. If this returns
+    null, throw an error.
+
+* Otherwise:
+  
+  * If `$module` is not a string, throw an error.
+
+  * Let `use` be the `@use` rule in [the current source file][] whose
+    namespace is equal to `$module`. If no such rule exists, throw an error.
+
+  * Return [`use`'s module][]'s mixin named `$name`, or throw an error if no
+    such mixin exists.
+
+  [the current source file]: ../spec/spec.md#current-source-file
+  [`use`'s module]: ../spec/at-rules/use.md#a-use-rules-module
+
+### `meta.module-mixins()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.module-mixins($module)
+```
+
+* If `$module` is not a string, throw an error.
+
+* Let `use` be the `@use` rule in [the current source file][] whose namespace is
+  equal to `$module`. If no such rule exists, throw an error.
+
+* Return a map whose keys are the quoted string names of mixins in
+  [`use`'s module][] and whose values are the corresponding mixins.
+
+### `meta.accepts-content()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.accepts-content($mixin)
+```
+
+* If `$mixin` is not a mixin, throw an error.
+
+* Return a boolean which is true if the body of the mixin has an `@content`
+  rule.
+
+## Mixins
+
+### `meta.apply()`
+
+```
+meta.apply($mixin, $args...)
+```
+
+* If `$mixin` is not a mixin, throw an error.
+
+* If the current `@include` rule has a `ContentBlock`, it should be passed down
+  to `$mixin`. Mixins must be invoked with `@include`, so the current include
+  rule is guaranteed to exist.
+
+* Execute the `ArgumentInvocation` `(...$args)` with `$mixin`'s
+  `ArgumentDeclaration` in `$mixin`'s scope.
+
+* Execute each statement in `$mixin`.
+
+## JavaScript API
+
+### Design Decisions
+
+Mixins differ from functions in that the result of their execution is a Sass AST
+node, and not a SassScript value. Sass today does not expose ways to create or
+manipulate AST nodes through the JavaScript API, nor does it intend to do so in
+the future.
+
+For this reason, it is not meaningful -- or even possible -- to construct or 
+execute a mixin through the JavaScript API. A mixin object shall be opaque, and
+the only operation available shall be to return the object as-is.
```

Comment:
This should be a subheading of `## Summary`.

Reply:
Implemented [here](https://github.com/sass/sass/pull/3626/commits/abbfecbce5b43608144ca8b37976f06fdec53763#diff-646077cb818e2f24ca0be3af54b487a8254fff79c51d33181a2e6c5483792f88R66). Since these are design decisions only about the JavaScript API, I updated the header and moved it under the toplevel `## Summary`.

I wasn't entirely sure if you meant that I should add a summary subheader to the JavaScript API header as well, or if the JavaScript API header should _also_ be moved under the main summary header. Please let me know if you had other thoughts here.